daisy/common.go

```diff
@@ -29,6 +29,9 @@ import (
 	"google.golang.org/api/compute/v1"
 )
 
+// FullResourceURLPrefix represents the prefix of full resource url
+var FullResourceURLPrefix = "https://www.googleapis.com/compute/[^/]*/"
```

Comment:
We can't support this style in daisy, we can only support the partial url as this changes based on the endpoint used. This is setup once using BasePath when creating the client, allowing for users to set this would give them the wrong idea as we will just throw it away and resources will be referenced based on the basepath

Reply:
Hi Andrew, per collected log, there are a couple of user errors caused by the full url input. So I tried to figure out whether it should be supported.
Per https://cloud.google.com/compute/docs/reference/rest/v1/instances --> networkInterfaces[].network, we can see compute API accepts this format. I think that is why users tried to inoput this format - they read the compute API doc.

I understand the versioning thing you mentioned, that is what I have considered. In this regex, the prefix "https://www.googleapis.com/compute/[any version]/" should be able to compatible with any version. After the prefix, it's the partial part, "projects/project/global/networks/network".
Your concern is, what if the next version changed the url pattern? However, if the pattern changed, it means our current supported partial url "projects/project/global/networks/network" also changed. So, it's the same thing.

On the other hand, compute API v1 supports the full url, which means usually there is no worry about user think the resource will be referenced based on the basepath.

Let me know if you disagree.

Reply:
Another option: preprocess the param in wrapper layer to remove the prefix. Keep daisy core simple and clean. How do you think?

Reply:
The issue is the "https://www.googleapis.com/compute/[any version]/" tells you about where the resource is, for example truing to create an image from "https://www.googleapis.com/compute/v1/" in "https://www.googleapis.com/compute/staging/" is not possible (complety different environment), and simply cutting that bit out is misleading. Daisy already has an option for BasePath, so its fine that we don't support full path elsewhere (and we shouldn't). I think it can make sense to support it in the wrapper if you want though since the use case for that tool is different that daisy and the contract with the user is different.

Reply:
Thanks for this guys, I appreciate you're probably thinking of Turkey, but have you come to an agreement above? 

The "shared VPC" pattern came recommended by google, which is why I'm trying to create the resource in project a, using the networking from the "shared vpc's" project, I appreciate it's a bit of an edge case. :/

Reply:
Hi Ben, we basically had an agreement.
Actually, We are not discussing your requirement but are trying to go even further to support more rarely-used URL format (starts with https://). Your 'projects/a-project/xxx' input will definitely be supported in this PR.

Reply:
Moved to wrapper layer @adjackura